Runtime/Model/Metrics/MetricsSubmissionQueue.cs

```diff
@@ -138,14 +138,18 @@ internal void SendPayload(ICollection<T> events, uint attempts = 0)
 
         public void SendPendingEvents(float time)
         {
-            for (int index = 0; index < _submissionJobs.Count; index++)
+            for (int index = 0; index < _submissionJobs.Count; )
             {
                 var submissionJob = _submissionJobs.ElementAt(index);
```

Comment:
I believe it's worth adding a null check here. Can we verify if the submissionJob is not equal to null ?